ui/src/app/pages/workspace/featured-workspaces.tsx

```diff
@@ -193,6 +193,7 @@ export const FeaturedWorkspaces = (props) => {
         );
       setWorkspaceList(
         workspacesReceived.items
+          .filter((fw) => fw.workspace.featuredCategory === currentTab.category)
```

Comment:
I'm curious if this will work, because it does not seem to remove the race condition. 

I believe that if you have two calls to two different featured category and the first call finishes after the second, you will end up in a weird spot. You will end up with `workspacesReceived` reflecting the results from the first call but `currentTab.category` reflecting the category of the second call. In this case, no results will be shown, because there shouldn't be any workspaces in the first call of the second category.

I wonder if you could rewrite this to something like this:
```
const [workspaceRequestAbortController, setWorkspaceRequestAbortController] =
    useState<AbortController|null>(null);
...
const controller = new AbortController();
const signal = controller.signal;
setWorkspaceRequestAbortController(controller);
featuredWorkspaceApi().getFeaturedWorkspacesByCategory(currentTab.category, { signal })
  .then(workspacesReceived => {
          setWorkspaceList(
        workspacesReceived.items
          .sort((a, b) => a.workspace.name.localeCompare(b.workspace.name))
          .map((w) => new WorkspacePermissions(w))
      );
  })
  
  
useEffect(() => {
    workspaceRequestAbortController.abort()
    props.hideSpinner();
    getAllPublishedWorkspaces();
  }, [currentTab]);
```